src/coord/src/persistence.rs

```diff
@@ -150,14 +164,43 @@ impl Source {
                 // We'll write down the data to a file with a `-tmp` prefix to
                 // indicate a write was in progress, and then atomically rename
                 // when we are done to indicate the write is complete.
-                let tmp_path = self.path.join(format!("{}-tmp", filename));
+                let tmp_path = self.path.join(format!("{}-tmp", file_name));
+                let path = self.path.join(file_name);
 
-                std::fs::write(&tmp_path, buf)?;
-                let final_path = self.path.join(filename);
-                std::fs::rename(tmp_path, final_path)?;
+                tokio::spawn(Self::flush(path, tmp_path, buf));
```

Comment:
I'm not familiar enough with the approach here to say for sure, but this maybe seems dangerous to me. Don't successive calls to flush need to be serialized? Otherwise you might persist a later chunk without an earlier chunk?

Reply:
my thought was that we could relax the serializability constraint because we can detect the case where we have a later chunk without an earlier chunk on restart, and if so, its not the end of the world, and we omit the part of the prefix thats incomplete? I am a bit concerned about blocking writes in the case where theres a lot of source data / many partitions but I'm also happy to defer this part of the change until we observe that being a problem in practice. the more important thing that I'd like to get merged is the user interface change

Reply:
I've gone ahead and removed the async-ness for now